app/assets/javascripts/modules/gtm-track-copy.js

```diff
@@ -0,0 +1,35 @@
+function GTMCopyListener ($module, dataLayer) {
+  this.$module = $module
+  this.dataLayer = dataLayer || window.dataLayer
+}
+
+GTMCopyListener.prototype.handleCopyEvent = function (component) {
+  this.dataLayer.push({
+    'event': 'text-copied',
+    'component': component
```

Comment:
If I remember right Andy was trying to track which copy-to-clipboard component was used so just the component name might not suffice as you'd not be able to differentiate between them.

We have a data attribute set on the input: https://github.com/alphagov/content-publisher/blob/a656889ebe093551a5a2d848cbc937bad2bee886/app/views/documents/show/_submitted_for_review.html.erb#L7

I wonder if we could send that rather than the component name?

Reply:
Sure, I thought we want to track the component overall and then narrow differentiate using the page they're being used on. Can update to use explicit gtm values for each usage.